runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftOverlay.java

```diff
@@ -78,35 +92,92 @@ public Dimension render(Graphics2D graphics)
 		for (WidgetItem item : widgetItems)
 		{
 			Varbits varbits;
+			int maxEssence;
+			boolean isBroken = false;
 
 			switch (item.getId())
 			{
 				case ItemID.SMALL_POUCH:
 					varbits = Varbits.POUCH_SMALL;
+					maxEssence = 3;
 					break;
 				case ItemID.MEDIUM_POUCH:
+					varbits = Varbits.POUCH_MEDIUM;
+					maxEssence = 6;
+					break;
 				case MEDIUM_POUCH_DAMAGED:
 					varbits = Varbits.POUCH_MEDIUM;
+					isBroken = true;
+					maxEssence = 6;
 					break;
 				case ItemID.LARGE_POUCH:
+					varbits = Varbits.POUCH_LARGE;
+					maxEssence = 9;
+					break;
 				case LARGE_POUCH_DAMAGED:
 					varbits = Varbits.POUCH_LARGE;
+					isBroken = true;
+					maxEssence = 9;
 					break;
 				case ItemID.GIANT_POUCH:
+					varbits = Varbits.POUCH_GIANT;
+					maxEssence = 12;
+					break;
 				case GIANT_POUCH_DAMAGED:
 					varbits = Varbits.POUCH_GIANT;
+					isBroken = true;
+					maxEssence = 12;
 					break;
 				default:
 					continue;
 			}
 
 			final Rectangle bounds = item.getCanvasBounds();
 			final TextComponent textComponent = new TextComponent();
-			textComponent.setPosition(new Point(bounds.x, bounds.y + 16));
-			textComponent.setText(String.valueOf(client.getSetting(varbits)));
+			int essenceInPouch = client.getSetting(varbits);
+
+			if (essenceInPouch == 0)
+			{
+				graphics.setColor(RED);
+			}
+			else if (essenceInPouch == maxEssence)
+			{
+				graphics.setColor(GREEN);
+			}
+			else
+			{
+				graphics.setColor(YELLOW);
+			}
+			graphics.fill(bounds);
+
+			if (isBroken)
+			{
+				BufferedImage image = itemManager.getImage(ItemID.BANK_FILLER);
+				if (image != null)
+				{
+					BufferedImage resizedImg = new BufferedImage(
+							IMAGE_SIZE.width,
+							IMAGE_SIZE.height,
+							BufferedImage.TYPE_INT_ARGB
+					);
+
+					Graphics2D g = resizedImg.createGraphics();
+					g.drawImage(image, 0, 0, IMAGE_SIZE.width, IMAGE_SIZE.height, null);
```

Comment:
Can you not just fillRect here instead? Regardless I would not fetch+resize the bank filler image per frame like this.